intranet/static/js/login.js

```diff
@@ -105,4 +105,23 @@ $(function() {
             });
         }
     });
+
+    function doneTyping () {
+        let re = /^(\d{4})?[A-z]+\d?$/;
```

Comment:
```suggestion
        let re = /^(\d{4})?[a-zA-Z]+\d?$/;
```

`[A-z]` will match everything in `range(ord("A"), ord("z"))`, which includes some extra characters, like `[`, `\`, and `]`. This is probably what Codacy was warning about.

Reply:
It didn't solve the issue, but it is fixed. Just scanning through a couple of google searches, it seems that this rule might be complaining about the use of parenthesis because when used as a capturing group, they can provide an avenue for a [ReDoS attack](https://en.wikipedia.org/wiki/ReDoS). I don't believe that is something we need to worry about in this case.